client/dashboard/components/settings/general/store-address.tsx

```diff
@@ -16,14 +16,21 @@ import { useSelect } from '@wordpress/data';
 import { getAdminSetting } from '~/utils/admin-settings';
 
 const { countries } = getAdminSetting( 'dataEndpoints', { countries: {} } );
+
+type Option = { key: string; label: string };
+
 /**
  * Check if a given address field is required for the locale.
  *
  * @param {string} fieldName Name of the field to check.
  * @param {Object} locale Locale data.
  * @return {boolean} Field requirement.
  */
-export function isAddressFieldRequired( fieldName, locale = {} ) {
+export function isAddressFieldRequired(
+	fieldName: string,
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
I am not very familiar with TypeScript. But, isn't there a way to handle this without using `any` and the corresponding `eslint-disable-next-line`?

Reply:
Thank you for bringing this up. I used `any` here because this is a complex object and there is no type defined in the data source. If we want to define a type for it, we need to trace the source but it is only used in this function, which I think it's not worthy.   However, I just changed this to `unknown` so we don't need the `eslint-disable-next-line`.